### DISCUSSION TITLE
How can I get a core dimension in a group?

### DISCUSSION THREAD

**Discussion by ricardobarroslourenco** [pydata/xarray#7636] [Q&A] [answered]
I'm processing the following data with `xr.apply_ufunc`:

![Screenshot from 2023-03-16 09-13-36](https://user-images.githubusercontent.com/7102375/225627855-cc23c4ce-8de0-4013-8d3c-196f538e6b89.png)

And when I pass:
```
result = xr.apply_ufunc(
    global_correlation,  # first the function
    lai_array['lai_north_masked_30_groupby_year_and_season_mean_detrended'],
    lai_array['temp_north_masked_30_groupby_year_and_season_mean_detrended'],
    input_core_dims=[['year'],['year']],
    vectorize=True,
    dask='parallelized',
    output_dtypes=['float']
)
```
I get the error:
`ValueError: operand to apply_ufunc has required core dimensions ['year'], but some of these dimensions are absent on an input variable: ['year']`

How can I refer to `year`, which is part of the `group` (together with `season`)? I was able to make it work in the version without traversing all coords, as well when I pass  `input_core_dims=[['group'],['group']]` (which gives me wrong results, since I want to calculate separately for each season).

**Comment by dcherian** [comment, marked as answer]
You'll need to `unstack("group")` first which should work OK with your current chunking.

It seems like you're stacking/unstacking to work around the limitation that Xarray only lets you group by a single variable.

Using flox will be better with dask, and more natural: https://flox.readthedocs.io/en/latest/intro.html#grouping-by-multiple-variables

**Reply by ricardobarroslourenco** [reply]
Thanks for the reference and the solution! Now it works beautifully and calculates fast.
